lib/multiaddr.js

```diff
@@ -3,13 +3,34 @@
  * @returns {string} Parsed URI, e.g. `http://127.0.0.1:80`
  */
 export function multiaddrToHttpUrl (addr) {
-  const [, hostType, hostValue, ipProtocol, port, scheme, ...rest] = addr.split('/')
+  const [multiAddr, httpPathMultiAddr] = addr.split('/http-path')
+  const [, hostType, hostValue, ...multiAddrParts] = multiAddr.split('/')
+  let scheme, path, rest, port
+  if (addr.includes('/http-path')) {
+    scheme = multiAddrParts.shift()
+    rest = multiAddrParts
+    try {
+      // Remove leading slash and parse URI-encoded path
+      // If the http-path is empty both `.../http-path/` and `.../http-path` are valid, see https://github.com/multiformats/multiaddr/blob/cab92e8e6da2e70c5f1b8aa59976e71e6922b392/protocols/http-path.md#usage
+      path = decodeURIComponent(httpPathMultiAddr.replace(/^\/+/, ''))
```

Comment:
I got sidetracked here when trying to understand the regular expression: Why are we removing a leading slash, if in your example above (`.../http-path/` and `.../http-path`) we were talking about trailing slashes?

Then I realized that this handling is only for the case where `httpPathMultiAddr === '/'`. Therefore I propose to make this explicit. On top, this line is confusing because it's doing mixing two responsibilities:

1. Decoding URI components
2. Turning `'/'` into `''`

```suggestion
      path = httpPathMultiAddr === '/'
        ? ''
        : decodeURIComponent(httpPathMultiAddr)
```

Reply:
It is more than that. To account for the `http-path` having to match `/http-path` and not only `/http-path/` we need to always remove the trailing `/` after `http-path` if it exists. `decodeURIComponent` will convert /SOME_PATH` into `//SOME_PATH` which is why I removed the leading `/` too. But it should be an easy fix.

Reply:
I addressed this in 25b53e6.